go/teams/handler.go

```diff
@@ -42,6 +83,87 @@ func HandleRotateRequest(ctx context.Context, g *libkb.GlobalContext, teamID key
 	})
 }
 
+func sweepOpenTeamResetMembers(ctx context.Context, g *libkb.GlobalContext,
+	team *Team, resetUsers []keybase1.TeamCLKRResetUser) (needRepoll bool, err error) {
+	// When CLKR is invoked because of account reset and it's an open team,
+	// we go ahead and boot reset readers and writers out of the team. Key
+	// is also rotated in the process (in the same ChangeMembership link).
+	defer g.CTrace(ctx, "sweepOpenTeamResetMembers", func() error { return err })()
+
+	// Go through resetUsers and fetch non-cached latest EldestSeqnos.
+	resetUserSeqnos := make(map[keybase1.UID]keybase1.Seqno)
+	for _, u := range resetUsers {
+		if _, found := resetUserSeqnos[u.Uid]; found {
+			// User was in the list more than once.
+			continue
+		}
+
+		arg := libkb.NewLoadUserArg(g).
+			WithNetContext(ctx).
+			WithUID(u.Uid).
+			WithPublicKeyOptional().
+			WithForcePoll(true)
+		upak, _, err := g.GetUPAKLoader().LoadV2(arg)
+		if err == nil {
+			resetUserSeqnos[u.Uid] = upak.Current.EldestSeqno
+		} else {
+			g.Log.CDebugf(ctx, "Could not load uid:%s through UPAKLoader: %s", u.Uid)
+		}
+	}
+
+	var needTeamReload bool
+
+	err = RetryOnSigOldSeqnoError(ctx, g, func(ctx context.Context, _ int) error {
+		if needTeamReload {
+			team2, err := Load(ctx, g, keybase1.LoadTeamArg{
+				ID:          team.ID,
+				Public:      team.ID.IsPublic(),
+				ForceRepoll: true,
+			})
+			if err != nil {
+				return err
+			}
+			team = team2
+		}
+		needTeamReload = true
+
+		changeReq := keybase1.TeamChangeReq{None: []keybase1.UserVersion{}}
+		for _, u := range resetUsers {
+			eldestSeqno, found := resetUserSeqnos[u.Uid]
+			if !found || eldestSeqno == u.MemberEldestSeqno {
+				// Either we couldn't load users EldestSeqno or we
+				// determined that they are not reset.
+				continue
+			}
+			uv := keybase1.NewUserVersion(u.Uid, u.MemberEldestSeqno)
```

Comment:
Could we get the to-be-removed eldestseqno from the team chain instead of from the server-supplied `u.MemberEldestSeqno`?

Reply:
Hmm in theory server could just send us UIDs and we could try to figure out the reset by our own. What do you think? I went with the simple way here because I didn't want to do another loop for when there are multiple versions of same user in the team. Server still can't lie to us because then `MemberRole` will return `NONE` and we will not remove anyone.

Reply:
The additional loop seems like the right thing to do if it's not terribly more complicated.

Here's the trick. If `bob` is at `bob%22` is a reader of the team but a tricky server sends you `u.MemberEldestSeqno = bob%33`. Then `eldestSeqno != u.MemberEldestSeqno` and you are tricked into removing `bob%22` from the team.

It seemed you like you were deriving benefit from the server supplied hint of eldestseqno for busting the uidmapper cache.

Reply:
but `bob%33` would have to be in the team, right? because of
```
			role, err := team.MemberRole(ctx, uv)
			if err != nil {
				continue
			}
			if role == keybase1.TeamRole_READER || role == keybase1.TeamRole_WRITER {
				changeReq.None = append(changeReq.None, uv)
			}
```
IIRC it MemberRole would return err=nil and role=keybase1.TeamRole_NONE.

Reply:
I'll do what you're suggesting because trusting the server less rather than more should be the goal, but I'm wondering if I really committed a security flaw in there or not.

Reply:
Ah good point! I'm convinced that in this case it's fine. In general I think it's safer to ignore the server. So whatever you want here.